### PR TITLE
fix torch dataloader warning. re non-writable numpy arrays.

### PR DESCRIPTION
**Pull request by orm011**
This change is [Reviewable](https://reviewable.io/reviews/mkornacker/pixeltable/66)

### CODE REVIEW

File: pixeltable/utils/pytorch.py [hunk 1]
```diff
@@ -70,15 +70,23 @@ def _unmarshall(self, k: str, v: Any) -> Any:
...
-                return np.array(im)
-            else:
-                assert self.image_format == "pt"
-                import torchvision  # pylint: disable = import-outside-toplevel
-                return torchvision.transforms.ToTensor()(im)
+                return arr
+
+            assert self.image_format == "pt"
+            import torchvision  # pylint: disable = import-outside-toplevel
+            return torchvision.transforms.ToTensor()(arr)
```
Comment (mkornacker) [inline pixeltable/utils/pytorch.py]: leave comment explaining why you're not passing in the pil image (which according to the docs is legal).

File: pixeltable/utils/pytorch.py [hunk 2]
```diff
@@ -123,7 +131,8 @@ def _to_column_dict(tab : pa.Table) -> Dict[str, Any]:
             for k in tab.column_names:
                 if self.column_types[k].is_array_type():
                     # treat array columns as numpy arrays to easily preserve numpy type
-                    column_dict[k] = tab.column(k).to_numpy()
+                    arr = tab.column(k).to_numpy()
```
Comment (mkornacker) [inline pixeltable/utils/pytorch.py]: why break out arr?

(or val, below)

**Review by orm011 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mkornacker/pixeltable/66)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @mkornacker)

*[a discussion](https://reviewable.io/reviews/mkornacker/pixeltable/66#gh-1457927274:-NoTOM7O9nLAuZNrIGjT:b-896fix) (no related file):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/mkornacker/pixeltable/pull/66#discussion_r1457927274) to the right spot — sorry!)*

leave comment explaining why you're not passing in the pil image (which according to the docs is legal).
</blockquote></details>

Done.

___
*[a discussion](https://reviewable.io/reviews/mkornacker/pixeltable/66#gh-1457928221:-NoTOMfkAYmZnBALoX0r:byjo81l) (no related file):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/mkornacker/pixeltable/pull/66#discussion_r1457928221) to the right spot — sorry!)*

why break out arr?

(or val, below)
</blockquote></details>

reversed. 

___

File: pixeltable/utils/pytorch.py (hunk 1 above)
Comment (mkornacker) [reply pixeltable/utils/pytorch.py]: why is this necessary?

**Review by orm011 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mkornacker/pixeltable/66)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @mkornacker)

*[a discussion](https://reviewable.io/reviews/mkornacker/pixeltable/66#gh-1457927274:-NoTTAVh82xuZQfSxz1U:bau5gle) (no related file):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

why is this necessary?
</blockquote></details>

I commented on the line calling np.array(). above ensures the array is always writable. That explains why I pass that.  
I don't know why sometimes things are not writable, but I'm ok ensuring this doesn't happen.

___

**Review by orm011 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mkornacker/pixeltable/66)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @mkornacker)

*[a discussion](https://reviewable.io/reviews/mkornacker/pixeltable/66#gh-1457927274:-NoTaIfFDth64sncl-5_:b-dtm6f7) (no related file):*
<details><summary><i>Previously, orm011 (Oscar Moll) wrote…</i></summary><blockquote>

I commented on the line calling np.array(). above ensures the array is always writable. That explains why I pass that.  
I don't know why sometimes things are not writable, but I'm ok ensuring this doesn't happen.
</blockquote></details>

I'll just reverse it since it doesnt trigger the warning, but its not clear when you can expect it to be writable and when you cannot. If we see the warning again we'll see. 

___

**Review by mkornacker — APPROVED** [review]

**Pull request merged**